pkg/performanceprofile/cmd/render/render.go

```diff
@@ -162,7 +190,40 @@ func render(inputDir, outputDir string) error {
 			if err != nil {
 				return err
 			}
+			klog.Info(fileName)
+		}
+	}
+
+	return nil
+}
+
+// genBootstrapWorkloadPinningManifests is used to generate the appropriate bootstrap workload pinning manifests
+// based on the cluster CPU Partitioning Mode and the given MachineConfigPool names.
+// The created manifests are the default state configs for the node which make no assumption for which CPU's are used for pinning.
+// This is done to ensure that when we generate the specific WorkloadPinning driven by the PerformanceProfile, we'll always have this
+// default state as a fall back if the user was to delete the PerformanceProfile by accident.
```

Comment:
I am not sure this will prevent accidents, because all objects generated here get ownershipReference that points to the PerfProfile. So deletion of the profile removes the MC as well (have you tested this scenario?).

Reply:
Right, in this situation, these are the preliminary configs and do not belong to a PerformanceProfile, they're there to tell kubelet to correctly identify itself for CPUPinning

When the user creates a PerformanceProfile, that will generate an owned config that is tied to that performance profile which overrides the bootstrap configs. So when they delete that PerformanceProfile, the bootstrap config doesn't get touched and we revert back to the preliminary configs.

I have tested this behavior and it was consistent in reverting back the configs. What we might want to do in the future is decide if we want to introduce a static resource management component to the operator to enforce the preliminary configs more rigidly.

Reply:
@MarSik Just to clarify, since these files are bootstrap files with the intention of being user defined later, we give them a lower priority in the MachineConfig file hierarchy. So for a CPU Pinned cluster we always generate these regardless if the user supplied a PerformanceProfile during bootstrapping or not, since the PerformanceProfile will generate it's own config that will override the bootstrap ones.

Reply:
Oh, so this MC does not have the ownership references. I think that is an important piece of information to be mentioned in the code. This file stays in the cluster forever, right?

Reply:
Right, exactly. I'll add that information I explained here to the comment above.